Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.cpp

```diff
@@ -265,11 +279,38 @@ namespace AzToolsFramework
                             entity->GetId().ToString().c_str(),
                             m_templateSourcePath.c_str());
                     }
+
+                    // Take ownership of the entity to queue it for destruction
+                    orderedEntityDestructionList.push_back(AZStd::move(entity));
                 }
             }
 
+            // Helper, returns the depth of a given entity within its transform hierarchy
+            // A root entity with no parent has a depth of 0, its immediate descendents 1, its descendants' children 2, etc.
+            auto entityHierarchyDepth = [](AZ::EntityId id) -> int
+            {
+                AZ::TransformInterface* currentLevel = nullptr;
+                AZ::TransformBus::EventResult(currentLevel, id, &AZ::TransformBus::Events::GetParent);
```

Comment:
I gave blanket disabling the reparenting behavior (while prefabs are enabled) a try, and thus far everything seems OK in my manual testing - after some offline discussion it looks like the prefab system really shouldn't need the logic at all, so that's a win.